workspace.go

```diff
@@ -395,6 +410,12 @@ type VCSRepoOptions struct {
 	GHAInstallationID *string `json:"github-app-installation-id,omitempty"`
 }
 
+type WorkspaceSettingOverwritesOptions struct {
+	// using `json` rather than `json-api` so that nil values are truly omitted. in `jsonapi` the json key will still be present
```

Comment:
Wow, that is weird...then what does omitempty even do for json-api?

Reply:
I think this is more due to my choice of datatype in the schema. I can double check but I saw other parts of go-tfe where we use the same idea.

Reply:
`+1` on double checking this.

Reply:
🖖 Looks like on the majority of the nested objects we also avoid using json-api annotations, like [organization access options](https://github.com/hashicorp/go-tfe/blob/0ef3e908c2106bf5fc194c918da3985c10ffb9b6/team.go#L146-L158).

Seems this bug specifically applies to booleans in nested objects.